numpy/ma/testutils.py

```diff
@@ -124,7 +124,7 @@ def assert_equal(actual, desired, err_msg=''):
         for k, i in desired.items():
             if k not in actual:
                 raise AssertionError(f"{k} not in {actual}")
-            assert_equal(actual[k], desired[k], f'key={k!r}\n{err_msg}')
+            assert_equal(actual[k], i, f'key={k!r}\n{err_msg}')
```

Comment:
This change is not the one suggested by the latest version of  ruff (on my system at least). Instead of the change here we could also write this as
``` 
for k in desired.keys():
            if k not in actual:
                raise AssertionError(f"{k} not in {actual}")
            assert_equal(actual[k], desired[k], f'key={k!r}\n{err_msg}')
```

Reply:
I seem to recall no fixes were available here, or perhaps only "unsafe fixes". But yes, I went for a different fix that address the "performance" issue, perhaps at the expense of readability. I will follow your advice and use the above change instead.